File: stage2/s2_for_06.js
// S2_FOR_06 — Nested arrays
function sumNested(matrix) {
  let sum = 0;

  for (const row of matrix) {
    for (const value of row) {
      sum += value;
    }
  }

  return sum;
}

// Tests
console.log(
  sumNested([
    [1, 2],
    [3, 4],
  ]),
); // 10
console.log(sumNested([[5], [10, 15]])); // 30
console.log(sumNested([])); // 0

File: stage2/B/S2_FOR_06.py
# S2_FOR_06.py

def sum_nested(matrix):
    total = 0
    for row in matrix:
        if not isinstance(row, list):
            return None
        for n in row:
            total += n
    return total


def main():
    tests = [
        [[1, 2], [3, 4], [5]],
        [[-1, 1], [10]],
        [],
        [1, [2, 3]],  # invalid: first "row" isn't a list
    ]
    for t in tests:
        print(f"{t!r} -> {sum_nested(t)!r}")


if __name__ == "__main__":
    main()
